lib/galaxy/datatypes/registry.py

```diff
@@ -6,11 +6,12 @@
 import logging
 import os
 from string import Template
-from typing import Dict
+from typing import Dict, List, Optional, Tuple
 
 import yaml
 
 import galaxy.util
+from galaxy.model import DatasetInstance
```

Comment:
This could be moved under `if TYPE_CHECKING:` to reduce the chance of a future circular import issue